gdal/alg/marching_squares/segment_merger.h

```diff
@@ -60,7 +62,7 @@ struct SegmentMerger
         {
             for (auto it =lines_.begin(); it!=lines_.end(); ++it) {
                 if ( ! it->second.empty() )
-                    debug("remaining unclosed contour");
+                    std::cerr << "remaining unclosed contour" << std::endl;
```

Comment:
use CPLError(CE_Failure/CE_Warning, CPLE_AppDefined, ...) or CPLDebug("CONTOUR", ...) depending on the level of verbosity needed for this.
Same for other instances